aggregator/src/diagnostic.rs

```diff
@@ -0,0 +1,123 @@
+//! Writing diagnostic files to disk.
+
+use anyhow::Context;
+use derivative::Derivative;
+use janus_messages::{AggregationJobId, ReportMetadata};
+use std::{fmt::Debug, fs::File, io::Write, path::Path, time::SystemTime};
+use uuid::Uuid;
+
+/// Represents an illegal attempt to mutate an aggregation job.
+#[derive(Derivative, Clone)]
+#[derivative(Debug)]
+pub struct AggregationJobInitForbiddenMutationEvent {
+    /// The ID of the aggregation job.
+    pub aggregation_job_id: AggregationJobId,
+
+    /// The SHA-256 of the request that created the aggregation job.
+    #[derivative(Debug(format_with = "fmt_hash_option"))]
+    pub original_request_hash: Option<[u8; 32]>,
+
+    /// The ordered report metadatas from the request that created the aggregation job.
+    pub original_report_metadatas: Vec<ReportMetadata>,
+
+    /// The batch ID in the original request.
+    pub original_batch_id: String,
```

Comment:
nit: We could skip some extra quote marks, here and on the other batch ID, since this is already Debug formatted before the struct is constructed.
```suggestion
    #[derivative(Debug(format_with = "std::fmt::Display::fmt"))]
    pub original_batch_id: String,
```

Reply:
I think this is moot now because I switched the event serialization format to JSON. The `*_batch_id` fields end up looking like `"original_batch_id": "()",`